### PR TITLE
Fix #1109; extract useFilteredVersions composable; migrate Pagination to TS/Composable/CSS

### LINKED ISSUE #1109: Pagination of Version Pages is Broken
### Describe the bug

Clicking on the page buttons at the bottom of a mod page with many pages of versions doesn't work correctly.

1. It requires two clicks to open the actually correct page
2. Passing a page parameter in the URL doesn't work
3. Clicking on any page button if the current page is at least page 3 jumps back to page 1
4. Clicking the next button goes back to page 1
5. Clicking the back button goes back to page 1

Tested in latest Chrome 112 and Safari 16.3 on macOS.

### Steps to reproduce

Step 0 of each problem: Go to a mod page with many versions like JEI https://modrinth.com/mod/jei/versions

#### For 1.
1. Click the button for page 4
2. Observe that nothing happens
3. Click it again
4. Page 4 opens

#### For 2.
0. Open https://modrinth.com/mod/jei/versions?p=4
1. It doesn't open page 4

#### For 3.
1. Go to page 3 by clicking on it twice
2. Click on page 4
3. It goes back to page 1

#### For 4.
1. Click the next button
2. Nothing happens
3. Sometimes it does switch to page 2, but then goes back to page 1

If done starting on page 3, it goes back to page 1.

#### For 5.
1. Go to page 3 by clicking it twice
2. Click the back button
3. It goes back to page 1

### Expected behavior

The pagination bar should behave as expected from a regular pagination bar:
- Clicking the page goes to the right page
- Passing an URL parameter should open the given page
- Clicking the next or back button moves one page

### Additional context

Screen recording of the various problems (heavily compressed to get within file size limit):

https://user-images.githubusercontent.com/17120022/233645396-c98a5ebc-1ba5-4050-b4ca-dfdb9a831279.mov

### PR DESCRIPTION
**Pull request by falseresync**
Fix #1109
(for automation)

### CODE REVIEW

File: composables/useFilteredVersions.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { Ref } from 'vue'
+
+interface Version {
+  game_versions: any[]
+  loaders: any[]
+  version_type: any
+}
+
+export default (unfilteredVersions: Ref<Version[]> | Version[], onQueryChange: () => void) => {
```
Comment (brawaru) [inline composables/useFilteredVersions.ts:9]: Why not named function?

```suggestion
export default function useFilteredVersions(
  unfilteredVersions: Ref<Version[]> | Version[],
  onQueryChange: () => void,
) {
```

Comment (falseresync) [reply composables/useFilteredVersions.ts:9]: No reason really, I just usually write anonymous functions with `export default`

Comment (Jamalam360) [reply composables/useFilteredVersions.ts]: Does eslint not enforce consistency on this?

Comment (falseresync) [reply composables/useFilteredVersions.ts]: It seems it doesn't, but it does remove the last comma

**Comment by falseresync** [comment]
Superseded by  #1181 and #1089

**Pull request closed without merging**